### PR TITLE
issue #346 + #326 

### LINKED ISSUE #326: Add input types to function arguments
As explained [here](https://stackoverflow.com/questions/2489669/how-do-python-functions-handle-the-types-of-the-parameters-that-you-pass-in), python lets you specify the data type of function arguments with "type hints." 

However these types will not raise errors if they are not satisfied. That is the second thing we should do.

### LINKED ISSUE #346: Suppress output in removeSmallCCs
In removeSmallCCs method [here](https://github.com/neurodata/brainlit/blob/develop/brainlit/preprocessing/image_process.py), it would be nice to have a `verbose` argument that suppresses the tqdm error bar.

### PR DESCRIPTION
**Pull request by sejalsrivastava1**
Fixed #346 and #326.
 
Added default verbose=True arg to removeSmallCCs method in image_process.py.

Added type annotations to all .py files in algorithms, feature_extraction, napari_viterbrain, utils, viz, and preprocessing.

### CODE REVIEW

File: brainlit/algorithms/trace_analysis/fit_spline.py [hunk 1]
```diff
@@ -17,6 +19,9 @@
 Geometric Graph class
 """
 
+# sejalsrivastava1 added type annotations
```
Comment (tathey1) [inline brainlit/algorithms/trace_analysis/fit_spline.py]: any reason why we should keep these comments?

File: brainlit/algorithms/trace_analysis/fit_spline.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
+from typing import Tuple
 import numpy as np
+import pandas
```
Comment (tathey1) [inline brainlit/algorithms/trace_analysis/fit_spline.py]: let's do `import pandas as pd` (which we do elsewhere in the package)

Comment (tathey1) [reply brainlit/algorithms/trace_analysis/fit_spline.py]: then you'll need to change calls to pandas as pd (e.g. line 34)

File: brainlit/feature_extraction/base.py [hunk 3]
```diff
@@ -11,6 +12,8 @@
 from joblib import Parallel, delayed
 from typing import Optional, List, Union, Tuple
 
+# sejalsrivastava1 added type annotations
```
Comment (tathey1) [inline brainlit/feature_extraction/base.py]: shoould we remove?

File: brainlit/feature_extraction/base.py [hunk 4]
```diff
@@ -166,15 +169,15 @@ def fit(
...
-    ):
+        seg_ids: List[int],
+        ngl: NeuroglancerSession,
+        num_verts: int,
+        start_vert: int,
+        include_neighborhood: bool,
+        write: bool = False,
+        batch_size: int = None,
+        file_path: str = None,
+    ) -> None:
```
Comment (tathey1) [inline brainlit/feature_extraction/base.py:178]: I've never seen return type None (I assumed you just wouldn't put a return type), I'm intrigued by the idea though - any reason why you put this here? Have you seen it before?

File: brainlit/napari_viterbrain/viterbrain_plugin.py [hunk 5]
```diff
@@ -9,8 +10,10 @@
 import pathlib
 import napari
 
+# sejalsrivastava1 added type annotations
```
Comment (tathey1) [inline brainlit/napari_viterbrain/viterbrain_plugin.py]: remove?

File: brainlit/preprocessing/image_process.py [hunk 6]
```diff
@@ -188,7 +190,9 @@ def getLargestCC(segmentation: np.ndarray) -> np.ndarray:
     return largestCC
 
 
-def removeSmallCCs(segmentation: np.ndarray, size: Union[int, float]) -> np.ndarray:
+def removeSmallCCs(
```
Comment (tathey1) [inline brainlit/preprocessing/image_process.py:191]: can you please add verbose=False to this line: https://github.com/neurodata/brainlit/blob/704a0cc439e580fc6b3d4e5aa0812f20be6a67d6/brainlit/algorithms/generate_fragments/state_generation.py#L214

File: brainlit/algorithms/trace_analysis/fit_spline.py (hunk 1 above)
Comment (tathey1) [reply brainlit/algorithms/trace_analysis/fit_spline.py]: i'm leaning towards removing them (they are on several files). GitHub automatically keeps track of authors of different files, so I am not sure that adding these author comments are necessary. what do youo think?

**Review by tathey1 — CHANGES_REQUESTED** [review]
added some comments. lastly, i know you referenced the issues in the PR title, but can you link them and also add brief text summarizing what you did in the PR description? also, what do you mean by "commented some minor issues"?

File: brainlit/algorithms/trace_analysis/fit_spline.py (hunk 1 above)
Comment (sejalsrivastava1) [reply brainlit/algorithms/trace_analysis/fit_spline.py]: yep, I'll be removing them now -- they were mostly just for me to see while i worked in my own environment!

File: brainlit/feature_extraction/base.py (hunk 4 above)
Comment (sejalsrivastava1) [reply brainlit/feature_extraction/base.py:178]: I think I saw it somewhere else in the code and copied it to keep consistency, but I can remove if it's unnecessary.

File: brainlit/algorithms/trace_analysis/fit_spline.py (hunk 1 above)
Comment (sejalsrivastava1) [reply brainlit/algorithms/trace_analysis/fit_spline.py]: I think the issues are linked but I'll add them in the description. Also, I think I was just referencing comments I added asking if I should import some packages (ex. pandas) where they were needed, but I ended up just doing it and removing the comments.

File: brainlit/feature_extraction/base.py (hunk 4 above)
Comment (sejalsrivastava1) [reply brainlit/feature_extraction/base.py:178]: https://github.com/neurodata/brainlit/blob/46ecbd5c9a6fddd3fdf3622683a63a674caf43a0/brainlit/algorithms/connect_fragments/viterbrain.py#L247-L249

like here!

Comment (tathey1) [reply brainlit/feature_extraction/base.py:178]: cool lets keep it then

**Review by tathey1 — APPROVED** [review]
looks good

**Pull request merged**